ca/client.go

```diff
@@ -346,6 +346,29 @@ func (c *Client) Renew(tr http.RoundTripper) (*api.SignResponse, error) {
 	return &sign, nil
 }
 
+// Revoke performs the revoke request to the CA and returns the api.RevokeResponse
+// struct.
+func (c *Client) Revoke(req *api.RevokeRequest) (*api.RevokeResponse, error) {
```

Comment:
can you explain why you think another method would be better? I feel like the api is cleaner this way (while the implementation is lightly more complicated.

Reply:
Client does not enforce you to use a client certificate, so if you initialize this with just the root certificate, you won't be able to reuse that client, and you have to create a new one for each one for revoke. And it might be normal to initialize the client at the start of the server and have it in the context of your server so you don't have to look for the initialization parameters again.

Reply:
Basically, a simple method for self revoking would be good, you can even simplify that so you can automatically create the api.RevokeRequest

Reply:
I think it might make more sense to just add another parameter (rather than having another method like RevokeSelf) because the other method will still need to take a request body as an argument ({Passive: true}). So I'm thinking if the transport is non-nil then create a new client, otherwise, use the existing one. What do you think? In the mean time I've implemented it as a new method, but, it's just copy-pasta.

Reply:
I was proposing a self-revocation method, with just the transport, because I don't see a use case for self-revocating itself using only the `passive: true` option, right now it's the only thing that we support, but we would update that one in the future. 
If you don't like that option, I would prefer the API as you have it right now instead of adding an extra transport parameter to revoke, as you can always create a new client. It would be just a known issue.